### PR TITLE
[12.0][MIG] partner_industry_secondary: Migration to v12

### PR DESCRIPTION
**Pull request by cristinamartinrod**
cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please check the change about how parent_store works:

`parent_path = fields.Char(index=True)`

**Comment by pedrobaeza** [comment]
I have added full instructions about that in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0#tasks-to-do-in-the-migration

**Comment by cristinamartinrod** [comment]
Change done 👍🏻

**Comment by yajo** [comment]
I think this is still lacking the migration script.

**Review by tarteo — COMMENTED** [review]
Code LGTM. But I can't really say because the tests are not running.

**Review by pedrobaeza — DISMISSED** [review]
Migration script is missing

File: partner_industry_secondary/models/res_partner_industry.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# Copyright 2016 Tecnativa S.L. - Pedro M. Baeza
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, exceptions, fields, models, _
+
+
+class ResPartnerIndustry(models.Model):
+    _inherit = 'res.partner.industry'
+    _order = "parent_left"
```
Comment (HviorForgeFlow) [inline partner_industry_secondary/models/res_partner_industry.py]: This should be changed or removed. 
I've got this error when trying to open a Partner in Contacts
```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 654, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 312, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 696, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 344, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 337, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 939, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 517, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 962, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 954, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 749, in call_kw
    return _call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 736, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/models.py", line 2766, in read
    self._read_from_database(stored, inherited)
  File "/.repo_requirements/odoo/odoo/models.py", line 2927, in _read_from_database
    field.read(fetched)
  File "/.repo_requirements/odoo/odoo/fields.py", line 2656, in read
    order_by = comodel._generate_order_by(None, wquery)
  File "/.repo_requirements/odoo/odoo/models.py", line 4055, in _generate_order_by
    order_by_elements = self._generate_order_by_inner(self._table, order_spec, query)
  File "/.repo_requirements/odoo/odoo/models.py", line 4022, in _generate_order_by_inner
    raise ValueError(_("Sorting field %s not found on model %s") % (order_field, self._name))
ValueError: Sorting field parent_left not found on model res.partner.industry
```

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]

File: partner_industry_secondary/models/res_partner_industry.py (hunk 1 above)
Comment (cristinamartinrod) [reply partner_industry_secondary/models/res_partner_industry.py]: Done!

File: partner_industry_secondary/migrations/12.0.1.0.0/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
```
Comment (pedrobaeza) [inline partner_industry_secondary/migrations/12.0.1.0.0/post-migration.py]: Remove this copyright

File: partner_industry_secondary/models/res_partner_industry.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 class ResPartnerIndustry(models.Model):
     _inherit = 'res.partner.industry'
-    _order = "parent_left"
+    _order = "name"
```
Comment (pedrobaeza) [inline partner_industry_secondary/models/res_partner_industry.py]: Put `parent_path`

**Review by yajo — APPROVED** [review]

**Review by HviorForgeFlow — APPROVED** [review]

**Comment by rafaelbn** [comment]
@cristinamartinrod please review comments. My functional review is :+1:

**Review by rafaelbn — APPROVED** [review]
Tested functional :+1: 

Please travis should be :green_apple:

**Review by AdriaGForgeFlow — APPROVED** [review]
Code review and functional test LGTM :+1:

**Comment by cristinamartinrod** [comment]
All changes done!

File: partner_industry_secondary/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    industry_id = fields.Many2one(string='Main Industry')
+
+    secondary_industry_ids = fields.Many2many(
+        comodel_name='res.partner.industry', string="Secondary Industries",
+        domain="[('id', '!=', industry_id)]")
+
+    @api.constrains('industry_id', 'secondary_industry_ids')
+    def _check_industries(self):
+        if self.industry_id in self.secondary_industry_ids:
+            raise exceptions.UserError(
```
Comment (MiquelRForgeFlow) [inline partner_industry_secondary/models/res_partner.py]: I think this should be a ValidationError

**Review by MiquelRForgeFlow — APPROVED** [review]
Please, rebase

File: partner_industry_secondary/models/res_partner_industry.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2015 Antiun Ingenieria S.L. - Javier Iniesta
```
Comment (nikul-serpentcs) [inline partner_industry_secondary/models/res_partner_industry.py:1]: @cristinamartinrod I think here to add the only company name and  Contributors move  `CONTRIBUTORS` file

```
# Copyright 2015 Antiun Ingenieria S.L.
# Copyright 2019 Tecnativa S.L.
# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
```

Improve in all file

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1: 

not blocking, Improvements

File: partner_industry_secondary/models/res_partner.py (hunk 4 above)
Comment (cristinamartinrod) [reply partner_industry_secondary/models/res_partner.py]: Done!

**Comment by cristinamartinrod** [comment]
@pedrobaeza can you review again, please? Thanks :)

File: partner_industry_secondary/models/res_partner_industry.py (hunk 5 above)
Comment (yajo) [reply partner_industry_secondary/models/res_partner_industry.py:1]: Copyright and contributors are not exactly the same.

**Pull request merged**